application/contract/rootdomain/rootdomain.go

```diff
@@ -97,12 +97,15 @@ func (rd *RootDomain) AddBurnAddress(burnAddress string) error {
 	return nil
 }
 
-func (rd RootDomain) GetBurnAddress() (string, error) {
+func (rd *RootDomain) GetBurnAddress() (result string, err error) {
 	if len(rd.FreeBurnAddresses) == 0 {
-		return "", fmt.Errorf("no more burn address left")
+		return "", fmt.Errorf("no more burn addresses left")
 	}
 
-	return rd.FreeBurnAddresses[0], nil
+	result = rd.FreeBurnAddresses[0]
+	rd.FreeBurnAddresses = rd.FreeBurnAddresses[1:]
```

Comment:
What if we get error while adding new member to maps? Should we return burn address to list of free addresses in that case?

Reply:
Yes. I created rollBack func in this PR https://github.com/insolar/insolar/pull/1966